src/main/java/org/truffleruby/parser/ast/SymbolParseNode.java

```diff
@@ -57,12 +57,11 @@
     public SymbolParseNode(SourceIndexLength position, String name, Encoding encoding, CodeRange cr) {
         super(position);
         this.name = name;  // Assumed all names are already intern'd by lexer.
-
-        if (encoding == USASCIIEncoding.INSTANCE || cr == CodeRange.CR_7BIT) {
-            encoding = USASCIIEncoding.INSTANCE;
+        Rope str = StringOperations.encodeRope(name, encoding, CodeRange.CR_UNKNOWN);
```

Comment:
So this ignores the `cr` argument because it's not correct IIRC?
Can you add a comment about it?

Reply:
It seems a bit cheaper/cleaner to do:
```java
if (encoding.isAsciiCompatible() && StringOperations.isASCIIOnly(name)) {
  this.rope = StringOperations.encodeRope(name, USASCIIEncoding.INSTANCE, CodeRange.CR_7BIT);
} else {
  this.rope = StringOperations.encodeRope(name, encoding, CodeRange.CR_UNKNOWN);
}
```

Reply:
Fixed and comment added.